tests/contrib/pytest_bdd/test_pytest_bdd.py

```diff
@@ -0,0 +1,208 @@
+import json
+import os
+
+import pytest
+
+from ddtrace import Pin
+from ddtrace.ext import test
+from tests.utils import TracerTestCase
+
+
+_SIMPLE_SCENARIO = """
+Feature: Simple feature
+    Scenario: Simple scenario
+        Given I have a bar
+        When I eat it
+        Then I don't have a bar
+"""
+
+
+class TestPytest(TracerTestCase):
+    @pytest.fixture(autouse=True)
+    def fixtures(self, testdir, monkeypatch):
+        self.testdir = testdir
+        self.monkeypatch = monkeypatch
+
+    def inline_run(self, *args):
+        """Execute test script with test tracer."""
+
+        class PinTracer:
+            @staticmethod
+            def pytest_configure(config):
+                if Pin.get_from(config) is not None:
+                    Pin.override(config, tracer=self.tracer)
+
+        return self.testdir.inline_run(*args, plugins=[PinTracer()])
+
+    def subprocess_run(self, *args):
+        """Execute test script with test tracer."""
+        return self.testdir.runpytest_subprocess(*args)
+
+    def test_pytest_bdd_scenario(self):
+        """Test that pytest-bdd traces scenario with all steps."""
+        self.testdir.makefile(
+            ".feature",
+            parameters="""
+                Feature: Parameters
+                    Scenario: Passing scenario
+                        Given I have 0 bars
+                        When I eat it
+                        Then I have -1 bars
+
+                    Scenario: Failing scenario
+                        Given I have 2 bars
+                        When I eat it
+                        Then I have 0 bar
+
+                    Scenario: Failing converter
+                        Given I have no bar
+                """,
+        )
+        py_file = self.testdir.makepyfile(
+            """
+            from pytest_bdd import scenarios, given, then, when, parsers
+
+            scenarios("parameters.feature")
+
+            BAR = None
+
+            @given(parsers.re("^I have (?P<bars>[^ ]+) bar$"))  # loose regex
+            def have_simple(bars):
+                global BAR
+                BAR = bars
+
+            @given(parsers.re("^I have (?P<bars>\\d+) bars$"), converters=dict(bars=int))
+            def have(bars):
+                global BAR
+                BAR = bars
+
+            @when("I eat it")
+            def eat():
+                global BAR
+                BAR -= 1
+
+            @then(parsers.parse("I have {bars:d} bar"))
+            def check_parse(bars):
+                assert BAR == bars
+
+            @then(parsers.cfparse("I have {bars:d} bars"))
+            def check_cfparse(bars):
+                assert BAR == bars
+            """
+        )
+        file_name = os.path.basename(py_file.strpath)
+        self.inline_run("--ddtrace", file_name)
+        spans = self.pop_spans()
+
+        assert len(spans) == 10
```

Comment:
is it enough to check the number of spans? 

Also, why 10?

Reply:
number of scenarios + number of steps